x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -766,6 +882,128 @@ private static ApplicationResourcePrivileges parseApplicationPrivilege(String ro
         return builder.build();
     }
 
+    public static final class RemoteIndicesPrivileges implements Writeable, ToXContentObject {
+        private final IndicesPrivileges indicesPrivileges;
```

Comment:
We can also use inheritance here, as well as for RemoteIndicesPrivileges.Builder and IndicesPrivileges.Builder but I would prefer to make that part of a separate refactor: it would fit nicely with [cleaning up parsing ](https://github.com/elastic/elasticsearch/pull/90614/files#r988041641). Especially tweaking the builder to cleanly support inheritance is a bigger change.

Reply:
Defer is OK with me.